projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/intrinsics/c/LLVMSignal.java

```diff
@@ -0,0 +1,245 @@
+
+            if (function.equals(LLVM_SIG_DFL)) {
+                Signal.handle(signal, SignalHandler.SIG_DFL);
+            } else if (function.equals(LLVM_SIG_IGN)) {
+                Signal.handle(signal, SignalHandler.SIG_IGN);
+            } else {
+                Signal.handle(signal, this);
+            }
+        }
+
+        @Override
+        public void handle(Signal arg0) {
+            /*
+             * https://en.wikipedia.org/wiki/Sigaction#Replacement_of_deprecated_signal.28.29
+             *
+             * Signal handlers installed by the signal() interface will be uninstalled immediately
+             * prior to execution of the handler.
+             *
+             * @note I made such a program, and I had to delete the handler manually, so this
+             * sentence is probably not true
+             */
+            callTarget.call();
+        }
+
+        public LLVMFunctionDescriptor getFunction() {
+            return function;
+        }
+
+    }
+
+    /**
+     * Handling conversation from Signal number to Signal Object in a clean way.
+     *
+     * documentation: http://man7.org/linux/man-pages/man7/signal.7.html
+     */
+    private enum Signals {
+
+        // POSIX.1-1990
+        SIG_HUP("HUP"),
+        SIG_INT("INT"),
+        SIG_QUIT("QUIT"),
+        SIG_ILL("ILL"),
+        SIG_ABRT("ABRT"),
+        SIG_FPE("FPE"),
+        SIG_KILL("KILL"),
+        SIG_SEGV("SEGV"),
+        SIG_PIPE("PIPE"),
+        SIG_ALRM("ALRM"),
+        SIG_TERM("TERM"),
+        SIG_USR1("USR1"),
+        SIG_USR2("USR2"),
+        SIG_CHLD("CHLD"),
+        SIG_CONT("CONT"),
+        SIG_STOP("STOP"),
+        SIG_TSTP("TSTP"),
+        SIG_TTIN("TTIN"),
+
+        // SUSv2 and POSIX.1-2001
+        SIG_BUS("BUS"),
+        SIG_POLL("POLL"),
+        SIG_PROF("PROF"),
+        SIG_SYS("SYS"),
+        SIG_TRAP("TRAP"),
+        SIG_URG("URG"),
+        SIG_VTALRM("VTALRM"),
+        SIG_XCPU("XCPU"),
+        SIG_XFSZ("XFSZ"),
+
+        // various other signals
+        SIG_IOT("IOT"),
+        SIG_EMT("EMT"),
+        SIG_STKFLT("STKFLT"),
+        SIG_IO("IO"),
+        SIG_CLD("CLD"),
+        SIG_PWR("PWR"),
+        SIG_INFO("INFO"),
+        SIG_LOST("LOST"),
+        SIG_WINCH("WINCH"),
+        SIG_UNUSED("UNUSED");
+
+        public static Signals decode(int code) {
+            for (Signals currentSignal : values()) {
+                if (currentSignal.signal() != null && currentSignal.signal().getNumber() == code) {
+                    return currentSignal;
+                }
+            }
+            return null;
+        }
+
+        private final Signal signal;
+
+        Signals(String signalName) {
+            Signal constructedSignal;
+            try {
+                constructedSignal = new Signal(signalName);
+            } catch (IllegalArgumentException e) {
+                // it seems like this signal is not available on your system
```

Comment:
Probably with ```LLVMLogger.info``` You have to note there are always some signals which a system doesn't support, so we have to evaluate this at runtime